### PR TITLE
Minor changes related to support for unix sockets

### PR DESCRIPTION
**Pull request by halter73**
- Minor style changes
- Changes to ServerAddress parsing to support the syntax suggested here: https://github.com/aspnet/KestrelHttpServer/pull/157#discussion_r37570739

### CODE REVIEW

**Comment by Daniel15** [comment]
Looks good to me :+1:

**Comment by halter73** [comment]
@lodejard

File: test/Microsoft.AspNet.Server.KestrelTests/ServerAddressFacts.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        [Theory]
+        [InlineData("")]
+        [InlineData("//noscheme")]
+        public void FromUriReturnsNullForSchemelessUrls(string url)
+        {
+            Assert.Null(ServerAddress.FromUrl(url));
+        }
+
+        [Theory]
+        [InlineData("://emptyscheme", "", "emptyscheme", 0, "")]
```
Comment (davidfowl) [inline test/Microsoft.AspNet.Server.KestrelTests/ServerAddressFacts.cs]: add localhost in here somewhere

Comment (halter73) [inline test/Microsoft.AspNet.Server.KestrelTests/ServerAddressFacts.cs]: Added below.

**Comment by halter73** [comment]
This completes https://github.com/aspnet/KestrelHttpServer/issues/156

**Pull request merged**